pkg/controller/case_test.go

```diff
@@ -0,0 +1,583 @@
+		return fmt.Errorf("binding with proper state not found, the existing binding status: %+v", lastBinding.Status)
+	}
+	return err
+}
+
+func (ct *ControllerTest) WaitForServiceInstanceRemoved() error {
+	var lastInstance *v1beta1.ServiceInstance
+	err := wait.PollImmediate(pollingInterval, pollingTimeout, func() (bool, error) {
+		instance, err := ct.scInterface.ServiceInstances(testNamespace).Get(testServiceInstanceName, v1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return true, nil
+		}
+		lastInstance = instance
+		return false, nil
+	})
+	if err == wait.ErrWaitTimeout {
+		return fmt.Errorf("the service instance still exists: %+v", lastInstance)
+	}
+	return err
+}
+
+func (ct *ControllerTest) WaitForReadyInstance() error {
+	condition := v1beta1.ServiceInstanceCondition{
+		Type:   v1beta1.ServiceInstanceConditionReady,
+		Status: v1beta1.ConditionTrue,
+	}
+
+	var lastInstance *v1beta1.ServiceInstance
+	err := wait.PollImmediate(pollingInterval, pollingTimeout, func() (bool, error) {
+		instance, err := ct.scInterface.ServiceInstances(testNamespace).Get(testServiceInstanceName, v1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting Broker: %v", err)
+		}
+		lastInstance = instance
+
+		for _, cond := range instance.Status.Conditions {
+			if condition.Type == cond.Type && condition.Status == cond.Status {
+				if condition.Reason == "" || condition.Reason == cond.Reason {
+					return true, nil
+				}
+			}
+		}
+		return false, nil
+	})
+	if err == wait.ErrWaitTimeout {
+		return fmt.Errorf("the instance is not ready, current status: %+v", lastInstance.Status)
+	}
+	return err
+}
+
+func (ct *ControllerTest) WaitForReadyBroker() error {
+	condition := v1beta1.ServiceBrokerCondition{
+		Type:   v1beta1.ServiceBrokerConditionReady,
+		Status: v1beta1.ConditionTrue,
+	}
+
+	var lastBroker *v1beta1.ClusterServiceBroker
+	err := wait.PollImmediate(pollingInterval, pollingTimeout, func() (bool, error) {
+		broker, err := ct.scInterface.ClusterServiceBrokers().Get(testClusterServiceBrokerName, v1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting Broker: %v", err)
+		}
+		lastBroker = broker
+
+		for _, cond := range broker.Status.Conditions {
+			if condition.Type == cond.Type && condition.Status == cond.Status {
+				if condition.Reason == "" || condition.Reason == cond.Reason {
+					return true, nil
+				}
+			}
+		}
+
+		return false, nil
+	})
+	if err == wait.ErrWaitTimeout {
+		return fmt.Errorf("the broker is not ready, current status: %+v", lastBroker.Status)
+	}
+	return err
+}
+
+func (ct *ControllerTest) WaitForClusterServiceClass() error {
+	return wait.PollImmediate(pollingInterval, pollingTimeout, func() (bool, error) {
+		_, err := ct.scInterface.ClusterServiceClasses().Get(testClassExternalID, v1.GetOptions{})
+		if err == nil {
+			return true, nil
+		}
+
+		return false, err
+	})
+}
+
+func (ct *ControllerTest) WaitForClusterServicePlan() error {
```

Comment:
I can see that u have totally different approaches in `Wait` functions, e.g. WaitForReadyBroker vs WaitForClusterServiceClass vs WaitForClusterServicePlan.

Please make it consistent.

Reply:
I also would like to know this.